src/bootstrap/bootstrap.py

```diff
@@ -12,9 +12,34 @@
 import sys
 import tarfile
 import tempfile
+import sqlite3
 
 from time import time
 
+# Acquire a lock on the build directory to make sure that
+# we don't cause a race condition while building
+# Lock is created in `project_root/lock.db`
+def acquire_lock(root_dir):
+    path = os.path.join(root_dir, "lock.db")
```

Comment:
I'm not sure creating lock.db in the root is OK -- it feels a little messy. I think we should be able to create the build directory earlier, if it's not created yet, and place lock.db inside it.